src/serac/physics/utilities/functional/functional.hpp

```diff
@@ -128,29 +138,28 @@ class Functional<test(trial), execution_policy> : public mfem::Operator {
     SLIC_ERROR_IF(!G_trial_, "Couldn't retrieve element restriction operator for trial space");
 
     // Ensure the mesh has the appropriate neighbor information before constructing the face restriction operators
-    test_fes->ExchangeFaceNbrData();
-    trial_fes->ExchangeFaceNbrData();
-
-    // Generate the face restriction operators using the shared face mesh data
-    G_test_boundary_  = GetFaceRestriction<test>(test_fes);
-    G_trial_boundary_ = GetFaceRestriction<trial>(trial_fes);
-
-    input_L_.SetSize(P_trial_->Height(), mfem::Device::GetMemoryType());
-    input_E_.SetSize(G_trial_->Height(), mfem::Device::GetMemoryType());
-
-    output_E_.SetSize(G_test_->Height(), mfem::Device::GetMemoryType());
-    output_L_.SetSize(P_test_->Height(), mfem::Device::GetMemoryType());
+    if (test_space_) test_space_->ExchangeFaceNbrData();
```

Comment:
Minor/style: I think we try to put even single-statement things inside curly braces